app/modules/intelligence/agents/chat_agents/code_changes_chat_agent.py

```
@@ -29,58 +22,56 @@
-        self.mini_llm = mini_llm
-        self.llm = llm
+    def __init__(self, db: Session):
+        self.db = db
         self.history_manager = ChatHistoryService(db)
         self.prompt_service = PromptService(db)
         self.agents_service = AgentsService(db)
+        self.system_message = None
         self.chain = None
-        self.db = db
 
     @lru_cache(maxsize=2)
```

Ruff (0.8.2)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)